Monika After Story/game/event-rules.rpy

```diff
@@ -437,6 +438,72 @@ init -1 python:
             # returns the skip_visual boolean of the rule
             if rule:
                 return rule[0]
-                
+
             # False since there was no rule to check
             return False
+
+    class MASFarewellRule(object):
+        """
+        Static Class used to create farewell specific rules in tuple form.
+        That tuple is then stored in a dict containing this rule name constant.
+        Each rule is defined by a special random chance.
+        random_chance is used to define the 1 in random_chance chance that this
+        farewell can be called
+        """
+
+        @staticmethod
+        def create_rule(random_chance, ev=None):
+            """
+            IN:
+                random_chance - An int used to determine 1 in random_chance
+                    special chance for this farewell to appear
+                ev - Event to create rule for, if passed in
+                    (Default: None)
+
+            RETURNS:
+                a dict containing the specified rules
+            """
+
+            # random_chance can't be negative
+            if random_chance < 0:
+                raise Exception("random_chance can't be negative")
+
+            # return the rule inside a dict
+            rule = {EV_RULE_FAREWELL_RANDOM : random_chance}
+
+            if ev:
+                ev.rules.update(rule)
+
+            return rule
+
+
+        @staticmethod
+        def evaluate_rule(event=None, rule=None):
+            """
+            IN:
+                event - the event to evaluate
+                rule - the MASFarewellRule to check it's random_chance
+
+            RETURNS:
+                True if the random returned 1
+            """
+            # check if we have an event that contains the rule we need
+            # event rule takes priority so it's checked here
+            if event and EV_RULE_FAREWELL_RANDOM in event.rules:
+                rule = event.rules[EV_RULE_FAREWELL_RANDOM]
+
+            # sanity check if we don't have a rule return False
+            if rule is None:
+                return False
+
+            # store the rule for easy access
+            # keeping this so when we add another rule we'll keep
+            # the same code structure, also it's more readable changing the name
+            random_chance = rule
+
+            # check if random_chance is 0 return False
+            if random_chance == 0:

```

Comment:
Rather than check only 0, lets program this defensively and return False if random_chance is `<= 0`

Reply:
Alright I'll make that change, I did the check against 0 only because on the creation method I already raise an Exception if the random chance is negative. Should I adjust the greeting check rule too? because there I only check against 0 too.

Reply:
Yeah adjust the other one too

Reply:
Done, thanks a lot fot taking the time to review it!